### ISSUE TITLE
'Train Spam' button not moving message to 'Spam' folder

### ISSUE THREAD

**Issue by DXXS** [moisseev/rspamd-spamness#30]
Version 2.0.0a12

Pressing the button when viewing mail message appears to do nothing!?

[Was using Thunderbird 102.5.0, upgraded to 102.7.2]
Server: Exim 4.92, rspamd 3.1, dovecot 2.3.4.1 (f79e8e7e4)

No evidence of activity in mail.info, mail.log, syslog, exim4 (mainlog, rejectlog, paniclog)

David

**Comment by moisseev** [comment]
Did you configure destination account and folder in the add-on options?

**Comment by DXXS** [comment]
No, I was under the impression that I was just setting 'Spam' to be the destination 'for all accounts' by leaving the account blank & just setting the folder (and there was no error message, or any other response to 'save') (?).  Looks like it does move them over if I originally specify an account (but then if I open the 'per account' setting, all of them are set to that!?).

**Comment by DXXS** [comment]
I have several+ accounts, with more not set up in TB yet (& only the 'yahoo' one uses 'Bulk' instead of 'Spam'), the ability to set the default spam folder once would be preferable!

**Comment by moisseev** [comment]
> No, I was under the impression that I was just setting 'Spam' to be the destination 'for all accounts' by leaving the account blank & just setting the folder 

The account is mandatory.

If you need to move messages into a Junk folder of the current account you can use Thunderbird's `Junk` button.

> (and there was no error message, or any other response to 'save') (?).

I've tested this case on 102.7.2 and got error notification:

![image](https://user-images.githubusercontent.com/2275981/218023577-06385474-33e8-49ad-b7bb-26b0360944ad.png)

 Looks like it does move them over if I originally specify an account (but then if I open the 'per account' setting, all of them are set to that!?).

That is expected as it is **default** destination **for all** accounts. You can override it  per-account in the `per-account folder settings`.

> I have several+ accounts, with more not set up in TB yet (& only the 'yahoo' one uses 'Bulk' instead of 'Spam'), the ability to set the default spam folder once would be preferable!

I disagree. It shouldn't be the default behavior since the feature is not intended for such a scenario. This will confuse users.

**Comment by DXXS** [comment]
Leaving the account blank is not giving any error message at all on my system (M1 Mac, OSX 12.6), much less indicating 'mandatory' on my system. There appear to be other unrelated issues too; guess I should file a different report.  I do have a 'spam' email account, it's just that I haven't specifically enabled it, as I have sieve set up to auto-detect the 'Spam' folder movements.  I do have multiple different servers running rclone at the moment also...

Ah, my 'junk' button was just changing the flag; looks like there is an option that I can turn on in TB preferences to have it do the move also!  I was attempting to avoid using their junk filtering apparatus, as I have rclone's neural system up & trained.

FWIW, I would generally prefer that a menu item be available when multiple messages are selected; might be nice to have the train menu items available in the right-click menu.

Just my opinions as someone with more than three decades of commercial software development under his belt!

Thanks for your assistance.

**Comment by moisseev** [comment]
> Leaving the account blank is not giving any error message at all on my system (M1 Mac, OSX 12.6)

I don't have any Mac at hand to check. May it be that notifications from Thunderbird are not enabled in OSX preferences?

> I do have a 'spam' email account, it's just that I haven't specifically enabled it, as I have sieve set up to auto-detect the 'Spam' folder movements. I do have multiple different servers running rspamd at the moment also...

Well, it is doable to add option to set a `current account` as the default destination account.

> FWIW, I would generally prefer that a menu item be available when multiple messages are selected; might be nice to have the train menu items available in the right-click menu.

Both are the most wanted features, but that weren't so easy to implement. I'll check if it has changed.
